lib/rex/post/meterpreter/extensions/stdapi/fs/file.rb

```diff
@@ -123,28 +123,48 @@ def File.basename(*a)
 
   #
   # Expands a file path, substituting all environment variables, such as
-  # %TEMP%.
+  # %TEMP% on Windows or $HOME on Unix
   #
   # Examples:
   #    client.fs.file.expand_path("%appdata%")
   #    # => "C:\\Documents and Settings\\user\\Application Data"
+  #    client.fs.file.expand_path("~")
+  #    # => "/home/user"
+  #    client.fs.file.expand_path("$HOME/dir")
+  #    # => "/home/user/dir"
   #    client.fs.file.expand_path("asdf")
   #    # => "asdf"
   #
-  # NOTE: This method is fairly specific to Windows. It has next to no relation
-  # to the ::File.expand_path method! In particular, it does *not* do ~
-  # expansion or environment variable expansion on non-Windows systems. For
-  # these reasons, this method may be deprecated in the future. Use it with
-  # caution.
-  #
   def File.expand_path(path)
-    request = Packet.create_request(COMMAND_ID_STDAPI_FS_FILE_EXPAND_PATH)
+    if client.platform == 'windows'
+      request = Packet.create_request(COMMAND_ID_STDAPI_FS_FILE_EXPAND_PATH)
 
-    request.add_tlv(TLV_TYPE_FILE_PATH, client.unicode_filter_decode( path ))
+      request.add_tlv(TLV_TYPE_FILE_PATH, client.unicode_filter_decode( path ))
 
-    response = client.send_request(request)
+      response = client.send_request(request)
+
+      return client.unicode_filter_encode(response.get_tlv_value(TLV_TYPE_FILE_PATH))
+    else
+      # For unix-based systems, do some of the work here
+      # First check for ~
+      path_components = path.split(separator)
+      if path_components.length > 0 && path_components[0] == '~'
+        path_components[0] = '$HOME'
+        path = path_components.join(separator)
+      end
+
+      # Now find the environment variables we'll need from the client
+      env_regex = /\$(?:([A-Za-z0-9_]+)|\{([A-Za-z0-9_.]+)\})/
```

Comment:
```suggestion
      env_regex = /\$(?:([A-Za-z0-9_]+)|\{([A-Za-z0-9_]+)\})/
```
Correct me if I am wrong here but I don't think periods are allowed in environment variables?

Reply:
Fixed in ebd927da44f89faf96a4a256f1f9627c91c698e0